spec/lib/performance_platform/gateway/unique_users_spec.rb

```diff
@@ -0,0 +1,148 @@
+describe PerformancePlatform::Gateway::UniqueUsers do
+  let(:session_repository) { DB[:sessions] }
+
+  before do
+    DB[:sessions].truncate
+    Timecop.freeze(Date.new(2018, 2, 1))
+  end
+
+  after do
+    Timecop.return
+  end
+
+  context 'stats for a week' do
+    subject { described_class.new(period: 'week') }
+
+    context 'given no sessions' do
+      it 'returns stats with zero unique users' do
+        expect(subject.fetch_stats).to eq(
+          count: 0,
+          metric_name: 'unique-users',
+          period: 'week',
+        )
+      end
+    end
+
+    context 'given many sessions' do
+      before do
+        session_repository.insert(
+          username: 'bob',
+          start: Date.today
```

Comment:
Odd, is this working because the time component is `00:00:00`?  I'd expect the stat to only pick up sessions from the last week?  I.e. 7 days pervious, but not today.

Reply:
That's exactly how it works. Same mechanism fuels PHP app.